src/point_one/polaris/polaris_client.cc

```diff
@@ -8,8 +8,18 @@
 
 #include <iomanip>
 
-#if defined(POLARIS_NO_GLOG)
-#include <iostream>
+#if P1_NO_PRINT
+#  include <iostream>
+static std::ostream null_stream(0);
```

Comment:
it is worth chcking if this does a heap alloc

Reply:
No, best I can tell it does not. It ultimately calls `basic_ios::init(nullptr)`, which calls `ios_base::_M_init()` and then sets its `_M_streambuf = nullptr` and sets `_M_streambuf_state = badbit`. `_M_init()` does not appear to do any allocs (https://github.com/gcc-mirror/gcc/blob/master/libstdc%2B%2B-v3/src/c%2B%2B98/ios_locale.cc).